packages/ndla-ui/src/Frontpage/FrontpageMenuPortal.tsx

```diff
@@ -0,0 +1,238 @@
+        props.animationDirection === 'in'
+          ? animations.durations.normal
+          : animations.durations.fast};
+    }
+    animation-fill-mode: forwards;
+    @keyframes ${props => props.animationNameIn} {
+      0% {
+        opacity: 0;
+        background: ${colors.brand.light};
+        ${props => css`
+          transform: translate(
+              calc(-100vw + ${props.elementRect.fromX}px),
+              calc(-50vh + ${props.elementRect.fromY}px)
+            )
+            scale(${props.elementRect.fromScale});
+        `}
+      }
+      10% {
+        opacity: 1;
+      }
+      20% {
+        border-radius: 100%;
+      }
+      99% {
+        border-radius: 10%;
+        transform: translate(calc(-50vw), 0)
+          scale(${fullSizedCircle});
+      }
+      100% {
+        border-radius: 0;
+        transform: translate(-50vw, 0);
+      }
+    }
+    @keyframes ${props => props.animationNameOut} {
+      0% {
+        border-radius: 0;
+        transform: translate(calc(-50vw), 30vh)
+          scale(${fullSizedCircle});
+      }
+      50% {
+        border-radius: 100%;
+      }
+      90% {
+        opacity: 1;
+      }
+      100% {
+        opacity: 0;
+        ${props => css`
+          transform: translate(
+              calc(-100vw + ${props.elementRect.fromX}px),
+              calc(-50vh + ${props.elementRect.fromY}px)
+            )
+            scale(${props.elementRect.fromScale});
+        `}
+      }
+    }
+  }
+`;
+
+interface StyledContainerProps {
+  animationDirection: 'in' | 'out';
+}
+
+const StyledContainer = styled.div<StyledContainerProps>`
+  transform: translate(-50vw, 0);
+  display: ${props => props.animationDirection === 'in' ? 'flex' : 'none'};
+  flex-direction: column;
+  align-items: flex-end;
+  margin: 0 auto;
+  max-width: 980px;
+  max-width: 100%;
+  }
+`;
+
+const ScrollableContent = styled.div`
+  padding: ${spacing.normal} ${spacing.large} ${spacing.large};
+  ${mq.range({ until: breakpoints.tablet })} {
+    padding: ${spacing.normal} ${spacing.normal} ${spacing.large};
+  }
+  width: 100%;
+  max-width: 980px;
+  height: 100vh;
+  overflow-y: scroll;
+  display: flex;
+  flex-direction: column;
+  align-items: flex-end;
+  animation-delay: ${animations.durations.fast};
+  animation-fill-mode: forwards;
+  opacity: 0;
+  ${animations.fadeInBottom(animations.durations.normal, spacing.normal)}
+  animation-timing-function: ease;
+  > div {
+    width: 100%;
+  }
+`;
+
+const StyledButton = styled.button<StyledContainerProps>`
+  border: 0;
+  background: none;
+  > svg {
+    color: ${colors.brand.primary};
+    width: ${spacing.medium};
+    height: ${spacing.medium};
+    ${mq.range({ from: breakpoints.desktop })} {
+      width: ${spacing.large};
+      height: ${spacing.large};
+    }
+  }
+`;
+
+interface Props {
+  children: React.ReactNode;
+  onClosed: () => void;
+  onClose: () => void;
+  animationDirection: 'in' | 'out';
+  elementRect: any;
```

Comment:
brukte du ikke du noen type for rect lenger opp?

Reply:
Flyttet ut i types og importerer derfra de stedene den brukes